mantidproject/mantid
# Add Load Button to SANS GUI

**Description of work.**
Add a load button to the new SANS GUI. Upon click, will load selected rows (all rows if none selected) workspaces into MantidPlot.

Additionally, this PR adds a save_can button. When clicked the unsubtracted can and sample workspaces will be added to ADS.

**Report to:** [nobody]

**To test:**
- On MantidPlot, go ->Interfaces->SANS -> ISIS SANS v2 experimental
- There will be a load button near the top right of the main tab.
- Load in any accepted user and batch file and click load. The corresponding workspaces should be added to MantidPlot *without* any further reduction taking place (e.g. if you name the output workspace in the table _outputWS_, that should not appear in MantidPlot, as that is a product of the reduction).
- Copy/duplicate/create several rows of data to process. Select a combination of them and click load. The workspaces corresponding to the selected rows *only* should be in MantidPlot.

Created additional tests in `batch_process_runner_test` to check that the correct signals are being sent when workspaces are loaded.

**To test save_can functionality**: load user and batch file which contains a separate can workspace. Process with combinations of *save_can* and *use_optimizations*.
- When *save_can* is ticked: output workspaces for LAB_SAMPLE and HAB_SAMPLE in a group
- When *use_optimization* is ticked: output workspaces for LAB/HAB _NORM and _COUNTS in a group titled "optimizations". The unsubtracted can workspace should be in the same group as the samples.

**If saving to a file only:**
- Only raw data and optimization workspaces should be on ADS
- If save_can is ticked the sample should **not** be on ADS but the can **should** be. However can workspaces should be in the optimizations group to be kept out of the way and avoid user confusion.
- Check that can and sample are also saved out when save_can is checked

**If saving to both:**
- check that can and sample workspaces saved when save_can is checked.
- Workspaces on ADS should be the same as when using save mode _memory_



Fixes #23797 
Fixes #23773 
Fixes #24212

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
